enterprise/server/workflow/service/service.go

```diff
@@ -400,6 +400,16 @@ func (ws *workflowService) createActionForWorkflow(ctx context.Context, wf *tabl
 				// re-cloned each time.
 				{Name: "recycle-runner", Value: "true"},
 				{Name: "preserve-workspace", Value: "true"},
+				// Pass the repo URL to the executor so that it can try to assign this
+				// task to a runner which has already checked out the repo, as opposed
+				// to another repo owned by the org. This also allows the scheduler
+				// to route tasks with different repos independently, resulting in better
+				// load balancing and more efficient runner reuse.
+				{Name: "workflow-repo-url", Value: wd.RepoURL},
+				// Hint to BuildBuddy's task router that it should favor reusing nodes
+				// that previously executed the workflow, since hitting a "cold" node
+				// that has not yet executed the workflow can be very expensive.
+				{Name: "task-router-preferred-node-limit", Value: "max"},
```

Comment:
task-router-preferred-node-limit is a lot. Why this name in particular?

some other ideas:
 enable-executor-affinity = true
 sticky-executors = true
 etc

Reply:
i like `enable-executor-affinity`

sticky implies that they always match

Reply:
This is configuring the max length of the list that we store in Redis for the given task, which effectively determines how many of the probes can "stick". By default it's 1, meaning that if the 1 "preferred" executor is overloaded then we'll fall back to random probing. "Max" means that we use the max allowed limit (I set it to 6 because it's upperbounded anyway by the number of probes we send + number of probes that fail, which is usually 0).

Reply:
Talked offline, we'll go with `workflow-action` and the task router will increase the preferred node limit internally when it sees that.